keel-core/src/main/kotlin/com/netflix/spinnaker/keel/IntentActivityRepository.kt

```diff
@@ -48,6 +69,18 @@ interface IntentActivityRepository {
 
   fun getHistory(intentId: String): List<String>
 
+  fun logConvergence(intentConvergenceRecord: IntentConvergenceRecord)
+
+  fun getConvergenceLog(intentId: String): List<IntentConvergenceRecord>
+
+  /**
+   * Permalink to a specific log message, identified by timeMs
+   * @param intentId The ID of the intent that is being evaluated.
+   * @param timeMs The timestamp of the log message,
+   *  used as the unique identifier of the message.
+   */
+  fun getConvergenceLogEntry(intentId: String, timeMs: Long): List<IntentConvergenceRecord>
```

Comment:
Slightly confusing name ... get*Entry implies a single result but a `List` is returned?

Reply:
There shouldn't be more than one here. Changing to return the first object if there is more than one with the same timestamp and then log a message . This shouldn't happen, and if it does it indicates that something is wrong with the queue processing.